@here/harp-mapview/lib/ScreenProjector.ts

```diff
@@ -70,14 +79,43 @@ export class ScreenProjector {
      * @param {(Vector3Like)} source The source vector to project.
      * @param {THREE.Vector2} target The target vector.
      * @returns {THREE.Vector2} The projected vector (the parameter 'target') or undefined if
-     * outside the screen.
+     * outside of the near/far plane. The point may be outside the screen.
+     */
+    projectToScreen(
+        source: Vector3Like,
+        target: THREE.Vector2 = new THREE.Vector2()
+    ): THREE.Vector2 | undefined {
+        const p = this.projectVector(source, ScreenProjector.tempV3);
+        if (isInRange(p)) {
+            return this.ndcToScreen(p, target);
+        }
+        return undefined;
+    }
+
+    /**
+     * Test if the area around the specified point is visible on the screen.
+     *
+     * @param {(Vector3Like)} source The centered source vector to project.
+     * @param {(Number)} width The width of the area in NDC fractions.
+     * @param {(Number)} height The height of the area in NDC fractions.
+     * @param {THREE.Vector2} target The target vector.
+     * @returns {THREE.Vector2} The projected vector (the parameter 'target') or undefined if
+     * the area is completely outside the screen.
      */
-    projectOnScreen(
+    projectAreaToScreen(
         source: Vector3Like,
+        width: number,
+        height: number,
         target: THREE.Vector2 = new THREE.Vector2()
     ): THREE.Vector2 | undefined {
         const p = this.projectVector(source, ScreenProjector.tempV3);
-        if (isOnScreen(p)) {
+        if (
+            isInRange(p) &&
+            p.x + width * 0.5 >= -1 &&
+            p.x - width * 0.5 <= 1 &&
+            p.y + height * 0.5 >= -1 &&
+            p.y - height * 0.5 <= 1
```

Comment:
```suggestion
            p.x + width * 0.5 <= 1 &&
            p.x - width * 0.5 >= -1 &&
            p.y + height * 0.5 <= 1 &&
            p.y - height * 0.5 >= -1
```
Better check why the unit tests didn't catch this.

Reply:
The unit tests seem not to catch this. BUT, the code that you propose does not work, the icons near the border do disappear while they should be visible. I keep the modified version.

Reply:
I still don't get why is like that, but check my suggestion below.

Reply:
This code has been removed. I suggest that we measure the impact of this PR, and optimize it in another PR if necessary.